### PR TITLE
Created Dockerfile | Added .gitignore file to server directory

### PR DESCRIPTION
**Pull request by IvayloSlavchev**

### CODE REVIEW

**Review by petya0111 — DISMISSED** [review]
Looks good.

**Review by petya0111 — COMMENTED** [review]
Test it again.

**Review by petya0111 — COMMENTED** [review]
Did you test some of your endpoints to approve that all is working on the docker container?

File: server/auth/authorization.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        if(!err){
+            req.user = user;
+            next();
+        } else {
+            return res.status(403).json({ message: 'User not authenticated' });
+        }
+    });
+}
+
+app.post('/renewAccessToken', (req, res) => {
```
Comment (petya0111) [inline server/auth/authorization.js]: Please make endpoints with dashes or in a path style. 
And read specifications on how to make better endpoint naming.
https://blog.dreamfactory.com/best-practices-for-naming-rest-api-endpoints/

File: server/auth/authorization.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    const refreshToken = req.body.token;
+
+    if(!refreshToken && !refreshTokens.includes(refreshToken)) {
+        return res.status(403).send('User not authenticated');
+    }
+
+    jwt.verify(refreshToken, process.env.REFRESH_TOKEN_ACCESS, (err, user) => {
```
Comment (petya0111) [inline server/auth/authorization.js]: Please do not use camel case in the endpoint naming.

File: server/auth/authorization.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+app.post('/login', (req, res) => {
+    const { user } = req.body;
+
+    if(!user) {
+        return res.status(403).send(user);
+    }
+
+    let accessToken = jwt.sign(user, process.env.ACCESS_TOKEN_SECRET, { expiresIn: "20m" });
+    let refreshToken = jwt.sign(user, process.env.REFRESH_TOKEN_ACCESS, { expiresIn: "7d" });
```
Comment (petya0111) [inline server/auth/authorization.js]: I think the refresh token should have for example '1h' one hour, but not 7 days expiry...

**Review by petya0111 — CHANGES_REQUESTED** [review]
Some comments about API endpoint better naming conventions.

File: server/auth/authorization.ts [hunk 4]
```diff
@@ -1,18 +1,18 @@
...
 
-
-let refreshTokens = [];
+const app = express();
+let refreshTokens: any = [];
 app.use(express.json());
 dotenv.config();
 
-function auth(req, res, next){
+function auth(req: any, res: any, next: any){
```
Comment (petya0111) [inline server/auth/authorization.ts]: The request/response body can be made with a DTO object to have a more clear type of object, Typescript helps very well with that. 
Please use strictly the object and needed fields that should be used. 
`any` type is a very unknown thing and you should write clear types. Thank you :)

**Review by petya0111 — CHANGES_REQUESTED** [review]
Please use DTO objects to have known types in Typescript.

**Comment by TeoLucidWork** [comment]
@IvayloSlavchev , there are not unit tests here.

**Review by petya0111 — APPROVED** [review]
Great work @IvayloSlavchev

**Pull request merged**